api/types/resource.go

```diff
@@ -168,8 +167,8 @@ func (m *Metadata) Expiry() time.Time {
 }
 
 // SetTTL sets Expires header using realtime clock
-func (m *Metadata) SetTTL(clock clockwork.Clock, ttl time.Duration) {
-	expireTime := clock.Now().UTC().Add(ttl)
+func (m *Metadata) SetTTL(now time.Time, ttl time.Duration) {
+	expireTime := now.Add(ttl)
```

Comment:
you can pass `expireTime` as the argument instead of time+duration

Reply:
This would make `SetTTL` an exact copy of `SetExpiry`. I think we should instead deprecate `SetTTL`, but keep it as it was with the clock rather than `time.Time`. @russjones Is there any reason to keep `SetTTL` when we already have `SetExpiry`?